File: simul/matlab/Utilities/Multipath.m
function [hs] = Multipath(omega, dist, ampl_coeff)

% pkg load miscellaneous;
% c = physical_constant('speed of light in vacuum');

% c = physconst('LightSpeed');
c = 299792458;

multipaths = zeros( length(omega),length(dist) );

for f = 1:length(omega)
    for k = 1:length(dist)
        a = c / (2 * dist(k) * omega(f)); % amplitude from distance
        phi = dist(k) * omega(f) / c ; % phase from distance
        multipaths(f,k) = ampl_coeff(k) * a * exp( - 1i * phi );
    end
end

hs = sum(multipaths, 2);
hs=hs.';
end

File: simul/python/Utilities/Multipath.py
import numpy as np

def Multipath(omega: float, dist: np.ndarray, ampl_coeff: np.ndarray) -> float:
    c = 299792458
    multipaths = np.zeros((1, max(dist.shape)), np.csingle)
    for k in range(dist.size):
        a = c / (2 * dist[0, k] * omega)  # amplitude from distance
        phi = (dist[0, k] * omega) / c  # phase from distance
        multipaths[:, k] = ampl_coeff[0, k] * a * np.exp(-1j * phi)
    hs = np.sum(multipaths, 1)
    return hs.transpose()
